WcaOnRails/app/controllers/delegates_controller.rb

```diff
@@ -7,4 +7,50 @@ class DelegatesController < ApplicationController
   def stats
     @delegates = User.delegates.includes(:actually_delegated_competitions)
   end
+
+  def delegate_probation_data
+    respond_to do |format|
+      format.json do
+        @probation_roles = Role.where(group_id: UserGroup.where(group_type: "delegate_probation"))
+        @probation_users = {}
+        @probation_roles.each { |probation_role|
+          @probation_users[probation_role.user_id] = User.find_by_id(probation_role.user_id)
+        }
```

Comment:
This uses the "ugly" `each`-approach. You should use the more efficient version from the now-deleted `def probations` method (see our previous GitHub conversations)

Reply:
Oops yes sorry, I've changed it now. Hope it's good now.